src/http/server.rs

```diff
@@ -1171,3 +1171,362 @@ pub mod ws {
             AsyncAcceptor<U, esp_idf_hal::mutex::Condvar, EspHttpWsDetachedSender>;
     }
 }
+
+#[cfg(all(feature = "experimental", esp_idf_esp_https_server_enable))]
```

Comment:
Exposing a separate module means a lot of code duplication (see below). Also, it is a bit awkward (user has to choose upfront whether that's an HTTP or an HTTPS server and use different types) and - it is not symmetric with HTTP client and MQTT client, where whether an SSL or a plain connection is being used does not change the client type.

What would be a more straightforward setup is if the HTTPS support is *fused* within the existing HTTP server (which would then become an HTTPS server as well). 
- All the SSL-related options are merged *inside* the existing HTTP server configuration, rather than wrapping it. 
- There is just one type - `EspHttpServer` which can handle either HTTP or HTTPS
- Whether an HTTP or HTTPS server is started is decided based on the `Configuration` struct (are SSL parameters inside it configured?)

The above approach means a lot more `esp_idf_esp_https_server_enable` conditionals sprinkled throughout the code, but I think that's overall a better compromise.

I'm not sure whether that's possible, but the above might lead to the possibility where we have a single HTTP server instance (`EspHttpServer`, that is) which - if configured so - is capable of serving *both* HTTP and HTTPS connections.

Reply:
Removed separate module and merged HTTPS functionality into the existing type.

- `server_certificate: Option<&'static str>` and `pub private_key: Option<&'static str>` are now conditionally merged into the existing HTTP configuration.
- Whether `httpd_ssl_start()` or `httpd_start()` is used, is decided by `esp_idf_esp_https_server_enable`.
- It is still possible to start a HTTP server with `esp_idf_esp_https_server_enable` by ommiting `server_certificate` and or `server_certificate`.

Serving both HTTP and HTTPS from the same instance might be possible, but the esp-idf HTTPS [component](https://github.com/espressif/esp-idf/blob/master/components/esp_https_server/src/https_server.c) doesn't expose that functionality. Providing it would require a lot of reimplementing.